metric_learn/itml.py

```diff
@@ -145,16 +145,21 @@ class ITML_Supervised(ITML):
   def __init__(self, gamma=1., max_iter=1000, convergence_threshold=1e-3,
                num_labeled=np.inf, num_constraints=None, bounds=None, A0=None,
                verbose=False):
-    """Initialize the learner.
+    """Initialize the supervised version of `ITML`.
+
+    `ITML_Supervised` creates pairs of similar sample by taking same class
+    samples, and pairs of dissimilar samples by taking different class
+    samples. It then passes these pairs to `ITML` for training.
 
     Parameters
     ----------
     gamma : float, optional
         value for slack variables
     max_iter : int, optional
     convergence_threshold : float, optional
-    num_labeled : int, optional
-        number of labels to preserve for training
+    num_labeled : int, optional (default=np.inf)
+        number of labels to preserve for training. If np.inf (default),
```

Comment:
what is meant by "preserve for training" needs to be clarified?

Reply:
what about: "number of labeled points to keep for building pairs. Extra 
        labeled points will be considered unlabeled, and ignored as such.
        Use np.inf (default) to use all labeled points." ?

Reply:
In fact IMO this parameter is useless. I will open an issue